pandas/_libs/tslibs/conversion.pyx

```diff
@@ -81,7 +81,11 @@ TD64NS_DTYPE = np.dtype("m8[ns]")
 # ----------------------------------------------------------------------
 # Unit Conversion Helpers
 
-cdef int64_t cast_from_unit(object ts, str unit) except? -1:
+cdef int64_t cast_from_unit(
+        object ts,
+        str unit,
+        NPY_DATETIMEUNIT reso=NPY_FR_ns
```

Comment:
can you make the names explicitly e.g. in_unit/out_reso?  usually when we see a unit we immediately convert it to a reso, so its weird to see both in the same place